Update documentation regarding compatibility (#1562)
tensorflow/addons: README.md

import tensorflow_addons as tfa
```

### Python Op Compatility
TensorFlow Addons is actively working towards forward compatiblity with TF2.x. 
However there are still a few private API uses within the respository so at the moment 

> we can only gurentee compatibility with the version of TF which it was built against. 

Reviewer: ```suggestion
we can only gurentee compatibility with the version of TF which it was tested against. 
```

We don't want to confuse users with the building of custom ops.

Warnings will be emitted during TFA import if your TensorFlow version does not match 
what it was built against.

### C++ Op Compatibility
TensorFlow C++ APIs are not stable and thus we can only guarentee compatibility with the 
version TFA was built against. It is possible custom-ops will work with multiple 
versions of TF, but there is also a chance for segfault or other problematic crash.

> Warnings will be emitted during TFA import if your TensorFlow version does not match 

Reviewer: ```suggestion
Warnings will be emitted when loading a custom op if your TensorFlow version does not match 
```
